NotificationsLab/app/src/main/java/net/serkanbal/notificationslab/MainActivity.java

```diff
@@ -0,0 +1,69 @@
+package net.serkanbal.notificationslab;
+
+import android.app.Notification;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.graphics.BitmapFactory;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.support.v7.app.NotificationCompat;
+
+public class MainActivity extends AppCompatActivity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        ConnectivityManager connMgr = (ConnectivityManager) getSystemService(Context.CONNECTIVITY_SERVICE);
+        NetworkInfo networkInfo = connMgr.getActiveNetworkInfo();
+        if (networkInfo != null && networkInfo.isConnected()) {
+
+            Intent intent = new Intent(this, ResultActivity.class);
+            PendingIntent pendingIntent = PendingIntent.getActivity(this,
+                    (int) System.currentTimeMillis(), intent, 0);
+
+            NotificationCompat.Builder builder = new NotificationCompat.Builder(this);
+            builder.setSmallIcon(android.R.drawable.ic_media_play);
+            builder.setContentTitle("Network Connected");
+            builder.setContentText("Your network is working fine!");
+            builder.setPriority(Notification.PRIORITY_MAX);
+            NotificationCompat.BigPictureStyle bigPictureStyle = new NotificationCompat.BigPictureStyle();
+            bigPictureStyle.setSummaryText("Your network is working fine!");
+            bigPictureStyle.bigPicture(BitmapFactory.decodeResource(getResources(),
+                    R.drawable.network_connected)).build();
+            builder.setStyle(bigPictureStyle);
+            builder.setContentIntent(pendingIntent);
+
+            NotificationManager notificationManager =
+                    (NotificationManager) getSystemService(NOTIFICATION_SERVICE);
+            notificationManager.notify(1, builder.build());
```

Comment:
Rather than hardcoding `1`, use a different constant for each notification. Then in the second activity, you can just call `cancel()` on the id for the "connected" notification, whether or not it is actually showing. No need to check for connectivity again from that activity.